src/main/scala/io/iohk/ethereum/jsonrpc/JsonMethodsImplicits.scala

```diff
@@ -29,58 +29,84 @@ trait JsonMethodsImplicits {
   protected def encodeAsHex(input: BigInt): JString =
     JString(s"0x${input.toString(16)}")
 
-  protected def tryExtractUnformattedData(input: JString): Either[JsonRpcError, ByteString] =
-    tryExtractUnformattedData(input.values)
+  private def decode(s: String): Array[Byte] = {
+    val stripped = s.replaceFirst("^0x", "")
+    val normalized = if (stripped.length % 2 == 1) "0" + stripped else stripped
+    Hex.decode(normalized)
+  }
+
+  protected def extractAddress(input: String): Either[JsonRpcError, Address] =
+    Try(Address(input)).toEither.left.map(_ => InvalidAddress)
+
+  protected def extractAddress(input: JString): Either[JsonRpcError, Address] =
+    extractAddress(input.s)
+
+  protected def extractBytes(input: String): Either[JsonRpcError, ByteString] =
+    Try(ByteString(decode(input))).toEither.left.map(_ => InvalidParams())
+
+  protected def extractBytes(input: JString): Either[JsonRpcError, ByteString] =
+    extractBytes(input.s)
 
-  protected def tryExtractAddress(input: JString): Either[JsonRpcError, ByteString] =
-    tryExtractUnformattedData(input, AddressLength, s"Invalid length for address ${input.values}, expected $AddressLength bytes")
+  protected def extractHash(input: String): Either[JsonRpcError, ByteString] =
+    extractBytes(input).filterOrElse(_.length == 32, InvalidParams(s"Invalid value [$input], expected 32 bytes"))
 
-  protected def tryExtractBlockHash(input: JString): Either[JsonRpcError, ByteString] =
-    tryExtractUnformattedData(input, BlockHashLength, s"Invalid length for block hash ${input.values}, expected $BlockHashLength bytes")
+  protected def extractQuantity(input: JValue): Either[JsonRpcError, BigInt] =
+    input match {
+      case JInt(n) =>
+        Right(n)
 
-  protected def tryExtractUnformattedData(input: JString, requiredLength: Int, errorMsg: String): Either[JsonRpcError, ByteString] =
-    tryExtractUnformattedData(input.values) match {
-      case Right(bytes) if bytes.length == requiredLength => Right(bytes)
-      case Right(_) => Left(JsonRpcErrors.InvalidParams(errorMsg))
-      case Left(err) => Left(err)
+      case JString(s) =>
+        Try(BigInt(decode(s))).toEither.left.map(_ => InvalidParams())
```

Comment:
I'm wondering should we check here (or in `decode`) that `s` has no leading zeroes as mentioned in https://github.com/ethereum/wiki/wiki/JSON-RPC#hex-value-encoding?